_posts/2020-07-06-whats-so-super-about-super-tdma.md

```diff
@@ -0,0 +1,158 @@
+// display documentation
+
+println """
+Equilateral triangle network
+----------------------------
+Internode distance:     ${range} m
+Slot length:            ${(1000*slot).round()} ms
+Simulation time:        ${time} s"""
+
+///////////////////////////////////////////////////////////////////////////////
+// simulate schedule
+
+simulate time, {
+
+  def n = []
+  n << node('1', address: 1, location: [0, 0, 0])
+  n << node('2', address: 2, location: [range, 0, 0])
+  n << node('3', address: 3, location: [0.5*range, 0.866*range, 0])
+
+  n.eachWithIndex { n1, i ->
+    n1.startup = {
+      def phy = agentForService PHYSICAL
+      phy[Physical.DATA].frameLength = phy[Physical.CONTROL].frameLength
+      add new TickerBehavior(1000*slot, {
+        def slen = schedule[i].size()
+        def s = schedule[i][(tickCount-1)%slen]
+        if (s) phy << new TxFrameReq(to: s, type: Physical.DATA)
+      })
+    }
+  }
+
+}
+
+// display statistics
+println """TX:                     ${trace.txCount}
+RX:                     ${trace.rxCount}
+Offered load:           ${trace.offeredLoad.round(3)}
+Throughput:             ${trace.throughput.round(3)}"""
+```
+
+First we set the distance between the nodes to be 650 m. With sound speed in water to be approximately 1540 m/s, the slot length turns out to be 422 ms which is set as can be seen in the code. The simulation time is set to 15 minutes and the high-throughput schedule is given as an input. The rest is pretty simple. The three network nodes are deployed at appropriate coordinates to form an equilateral triangle. The frame duration of the DATA channel packet is set to be smaller than the time slot length, i.e., we need to make sure that the frame length and data rate is such that the packet duration is smaller than 422 ms. Once we set this up, the packets are transmitted from each node in a `TickerBehavior` as per the schedule. Now, let's run this and look at the result:
+
+```term
+Equilateral triangle network
+----------------------------
+Internode distance:     650 m
+Slot length:            422 ms
+Simulation time:        900 s
+TX:                     3198
+RX:                     3011
+Offered load:           1.375
+Throughput:             1.295
+
+1 simulation completed in 2.615 seconds
+```
+The network capacity as we learned earlier is 1.5 (see [1] for more details). Although, in this network, our offered load is less than that since the packet duration is slightly lesser than the time slot duration and hence they are not fully utilized. And therefore, we see a reduced throughput but still 29.5% larger than what can be achieved traditionally. The fact that the practical frame durations used in underwater networks are comparable to the propagation delays led to this gain in throughput. This is remarkable and we should aspire to harness such knowledge of propagation delays in networks when designing performant protocols for higher throughput.
```

Comment:
There's a bunch of assumptions about synchronization and clocks. It would seem naïve to assume perfect sync and not address how a practical system might work, at least at a conceptual level.

Reply:
I've now added common practical issues that need to be taken care of and some ways in which they might be addressed.